src/System.Reflection/tests/AssemblyNameTests.cs

```diff
@@ -251,16 +251,36 @@ public void Name_CurrentlyExecutingAssembly()
             Assert.StartsWith("System.Reflection.Tests", assemblyName.Name);
         }
 
+        [Fact]
+        public static void FullName_WithPublicKey()
+        {
+            AssemblyName assemblyName = new AssemblyName("MyAssemblyName, Version=1.0.0.0");
+            assemblyName.SetPublicKeyToken(typeof(object).Assembly.GetName().GetPublicKeyToken());
+            assemblyName.SetPublicKey(typeof(object).Assembly.GetName().GetPublicKey());
+
+            Assert.NotNull(assemblyName.GetPublicKeyToken());
+            Assert.NotNull(assemblyName.GetPublicKey());
+            Assert.StartsWith("MyAssemblyName, Version=1.0.0.0, PublicKeyToken=", assemblyName.FullName);
+        }
+
+        [Fact]
+        public static void FullName_VersionToString()
+        {
+            AssemblyName assemblyName = new AssemblyName("MyAssemblyName");
+            assemblyName.Version = new Version(1, 2, 0x1ffff, 4);
```

Comment:
You could just add to Version_TestData() instead of writing a new [Fact] for this.